docs/gguf.md

```diff
@@ -18,6 +18,53 @@ GGUF is a format based on the existing GGJT, but makes a few changes to the form
 
 The key difference between GGJT and GGUF is the use of a key-value structure for the hyperparameters (now referred to as metadata), rather than a list of untyped values. This allows for new metadata to be added without breaking compatibility with existing models, and to annotate the model with additional information that may be useful for inference or for identifying the model.
 
+### GGUF Naming Convention
+
+GGUF follow a naming convention of `<Model>-<Version>-<ExpertsCount>x<Parameters>-<Quantization>.gguf`
+
+The components are:
+1. **Model**: A descriptive name for the model type or architecture.
+2. **Version**: (Optional) Denotes the model version number, formatted as `v<Major>.<Minor>`
+    - If model is missing a version number then assume `v0.0` (Prerelease)
+3. **ExpertsCount**: Indicates the number of experts found in a Mixture of Experts based model.
+4. **Parameters**: Indicates the number of parameters and their scale, represented as `<count><scale-prefix>`:
+    - `T`: Trillion parameters.
+    - `B`: Billion parameters.
+    - `M`: Million parameters.
+    - `K`: Thousand parameters.
+5. **Quantization**: This part specifies how the model parameters are quantized or compressed.
+   - Floating Representation:
+     - `BF16`: [16-bit bfloat16](https://en.wikipedia.org/wiki/Bfloat16_floating-point_format) [Google Brain](https://en.wikipedia.org/wiki/Google_Brain) truncated form of 32-bit IEEE 754 (1 sign bit, 8 exponent bits, 7 fractional bits)
+     - `F32`: [32-bit IEEE 754](https://en.wikipedia.org/wiki/Single-precision_floating-point_format#IEEE_754_standard:_binary32) floats per weight (1 sign bit, 8 exponent bits, 23 fractional bits)
+     - `F16`: [16-bit IEEE 754](https://en.wikipedia.org/wiki/Half-precision_floating-point_format) floats per weight (1 sign bit, 5 exponent bits, 10 fractional bits)
+   - Quantization (Compression) formats:
+     - `Q<X>`: X bits per weight, where `X` could be `4` (for 4 bits) or `8` (for 8 bits) etc...
+     - Variants provide further details on how the quantized weights are interpreted:
+       - `_K`: k-quant models, which further have specifiers like `_S`, `_M`, and `_L` for small, medium, and large, respectively, if they are not specified, it defaults to medium.
+       - `_<num>`: Different approaches, with even numbers indicating the model weights as a scaling factor multiplied by the quantized weight and odd numbers indicating the model weights as a combination of an offset factor plus a scaling factor multiplied by the quantized weight.
+            - Even Number (0 or 2): `<model weights> = <scaling factor> * <quantised weight>`
+            - Odd Number (1 or 3): `<model weights> = <offset factor> + <scaling factor> * <quantised weight>`
```

Comment:
This is better to be called "File type" component instead of "Quantization" as it corresponds to the `enum ggml_ftype` and `enum llama_ftype` enums:

https://github.com/ggerganov/ggml/blob/a85921414daf830bcf5beb668f17ccd1efcbd372/include/ggml/ggml.h#L395-L421

It is an identifier that describes the contents of the GGUF file and gives information to the user-code about what tensor data types to be expected. The specific contents and types are determined by the user-code (i.e. the project reading the GGUF file) and can vary depending on the project needs

No need to provide details such as the meaning of `_S`, `_M` and `_L` as these are specifics of the `llama.cpp` project and not in general for GGUF

I wrote some more about the file type enums here: https://github.com/ggerganov/llama.cpp/pull/7075#issuecomment-2101189684